### PR TITLE
[file_selector_platform_interface] null safety stable release

### PR DESCRIPTION
**Pull request by ditman**
Required for the migration to stable null safety of file_selector.

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/file_selector/file_selector_platform_interface/pubspec.yaml [hunk 1]
```diff
@@ -3,23 +3,22 @@ description: A common platform interface for the file_selector plugin.
 homepage: https://github.com/flutter/plugins/tree/master/packages/file_selector/file_selector_platform_interface
 # NOTE: We strongly prefer non-breaking changes, even at the expense of a
 # less-clean API. See https://flutter.dev/go/platform-interface-breaking-changes
-version: 2.0.0-nullsafety.0
+version: 2.0.0
 
 dependencies:
   flutter:
     sdk: flutter
   meta: ^1.0.5
```
Comment (stuartmorgan) [inline packages/file_selector/file_selector_platform_interface/pubspec.yaml:11]: Should be 1.3.0 I believe.

File: packages/file_selector/file_selector_platform_interface/pubspec.yaml [hunk 2]
```diff
@@ -3,23 +3,22 @@ description: A common platform interface for the file_selector plugin.
...
+  http: ^0.13.0
+  plugin_platform_interface: ">=1.0.0 <3.0.0"
+  cross_file: ^0.3.0
 
 dev_dependencies:
   test: ^1.15.0
   flutter_test:
     sdk: flutter
-  mockito: ^5.0.0-nullsafety.5
   pedantic: ^1.8.0
```
Comment (stuartmorgan) [inline packages/file_selector/file_selector_platform_interface/pubspec.yaml:20]: 1.10.0

File: packages/file_selector/file_selector_platform_interface/pubspec.yaml [hunk 3]
```diff
@@ -3,23 +3,22 @@ description: A common platform interface for the file_selector plugin.
...
   meta: ^1.0.5
-  http: ^0.13.0-nullsafety.0
-  plugin_platform_interface: ^1.1.0-nullsafety.2
-  cross_file: ^0.3.0-nullsafety
+  http: ^0.13.0
+  plugin_platform_interface: ">=1.0.0 <3.0.0"
+  cross_file: ^0.3.0
 
 dev_dependencies:
   test: ^1.15.0
```
Comment (stuartmorgan) [inline packages/file_selector/file_selector_platform_interface/pubspec.yaml:17]: 1.16.3 (IIRC)

**Review by stuartmorgan — APPROVED** [review]
LGTM with nits. Not sure why these don't get flagged by analysis.

**Comment by ditman** [comment]
🤦 

> Not sure why these don't get flagged by analysis.

Maybe it's because they are not major version bumps, and pub is probably able to locate the preview version anyways?

**Comment by ditman** [comment]
Published

**Pull request merged**